### PR TITLE
Make JCE provider configurable

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
Resolves - https://github.com/wso2/product-is/issues/15428

Merge this PR after merging https://github.com/wso2/carbon-kernel/pull/3497

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -210,7 +215,7 @@
         <carbon.identity.framework.version>5.11.148</carbon.identity.framework.version>
         <carbon.identity.framework.imp.pkg.version.range>[5.0.0, 6.0.0)
         </carbon.identity.framework.imp.pkg.version.range>
-        <carbon.kernel.version>4.6.1</carbon.kernel.version>
+        <carbon.kernel.version>4.9.0-beta</carbon.kernel.version>
```
Comment (ashensw) [inline pom.xml]: We can use 4.9.0 here since it's released now.

Comment (ChanikaRuchini) [reply pom.xml]: fixed

File: components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java [hunk 2]
```diff
@@ -819,13 +821,27 @@ public static List<String> getAIALocations(X509Certificate cert) throws Certific
     private static OCSPReq generateOCSPRequest(X509Certificate issuerCert, BigInteger serialNumber)
             throws CertificateValidationException {
 
-        // Add provider BC
-        Security.addProvider(new BouncyCastleProvider());
         try {
+            String providerName = ServerConfiguration.getInstance().getFirstProperty(ServerConstants.JCE_PROVIDER);
```
Comment (ashensw) [inline components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java]: Can't we utilize the getJCEProvider method for this also?

Comment (ChanikaRuchini) [reply components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java]: Fixed

**Review by ashensw — APPROVED** [review]

**Pull request merged**